### DISCUSSION TITLE
Usage of nested form provider.

### DISCUSSION THREAD

**Discussion by Suman085** [react-hook-form/react-hook-form#5682] [Q&A] [answered]
I was scaffolding a project today. Found that if we nest the formproviders, the handleSubmit callback gets triggered for all the external formcontext callbacks whenever the internal form is submitted even though if the inner formprovider is inside of a modal.. Is this the intended behavior of the provider itself or the react hook formprovider?

Similar scenario has been created in the sandbox as well.
https://codesandbox.io/s/formprovider-i8yhn?file=/src/modal.js

Please give us your feedbacks,
Thanks

**Comment by JamesWong1999** [comment, marked as answer]
`provider` shouldn't be nested.
